pghoard/pghoard.py

```diff
@@ -779,15 +779,23 @@ def handle_site(self, site, site_config):
         if metadata and not os.path.exists(self.config["maintenance_mode_file"]):
             if site in self.delta_backup_failures:
                 retries = self.delta_backup_failures[site].retries
-                if retries > site_config["basebackup_delta_mode_max_retries"]:
+                bih = site_config.get("basebackup_interval_hours")
+                backup_interval = datetime.timedelta(hours=bih) if bih else None
+                last_failed_time = self.delta_backup_failures[site].last_failed_time
+                since_last_fail_interval = utc_now() - last_failed_time if last_failed_time else None
+                if metadata["backup-reason"] == BackupReason.requested:
+                    self.log.info("Re-trying delta basebackup. Backup was requested")
+                elif backup_interval and since_last_fail_interval and since_last_fail_interval > backup_interval:
```

Comment:
Need to discuss whether we need to consider backoff retry internal for this two new cases. Just keeping retrying without waiting a bit seems less meaningful to me.

Reply:
I would not do any backoff in case if the backup was requested manually, it's up to the user to decide if backup needs to be tried.
As for the second case, I am not quite sure it makes sense because if the backup used to fail and maxed out and is now failing, probably the problem did not go away. Thus I would not waste resources and money on object storage calls.

Reply:
Agree. Only let "requested" backup keep retrying.